stl/inc/xpolymorphic_allocator.h

```diff
@@ -296,8 +296,8 @@ namespace pmr {
         }
 
         template <class _Uty>
-        _CXX17_DEPRECATE_POLYMORPHIC_ALLOCATOR_DESTROY void destroy(_Uty* const _Ptr) noexcept /* strengthened */ {
-            _STD _Destroy_in_place(*_Ptr);
+        void destroy(_Uty* const _Ptr) noexcept /* strengthened */ {
+            _Ptr->~_Uty();

```

Comment:
Above, `delete_object()` is still calling `_STD _Destroy_in_place(*_Ptr);`, but WG21-P2875R4 is changing it to:

> *Effects:* Equivalent to:
> ```cpp
> destroy(p);
> deallocate_object(p);
> ```

We should change it accordingly.

Reply:
Thanks! IIUC the line in `delete_object()` was buggy when initially being implemented in #1311 and after the acceptance of LWG-3036.

Although the previous wording (LWG-3036) said `allocator_traits<polymorphic_allocator>::destroy(*this, p)`, the fallback branch of `allocator_traits::destroy` (using `destroy_at` since C++20) wasn't really taken as `polymorphic_allocator::destroy` wasn't removed. So destruction of arrays has never been supported.